frontend/pages/admin/IntegrationsPage/cards/Mdm/components/EditTeamModal.tsx

```diff
@@ -1,29 +1,83 @@
-import React from "react";
+import React, { useState, useContext, FormEvent } from "react";
+
+import { AppContext } from "context/app";
+import configAPI from "services/entities/config";
 
 import Modal from "components/Modal";
 import Button from "components/buttons/Button";
 
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+
 interface IEditTeamModal {
   onCancel: () => void;
-  onEdit: () => void;
+  currentDefaultTeamName: string | undefined;
 }
 
 const baseClass = "edit-team-modal";
 
-const EditTeamModal = ({ onCancel, onEdit }: IEditTeamModal): JSX.Element => {
+const EditTeamModal = ({
+  onCancel,
+  currentDefaultTeamName,
+}: IEditTeamModal): JSX.Element => {
+  // availableTeams: Array<ITeamSummary>
+  const { availableTeams } = useContext(AppContext);
+
+  const teamNameOptions = availableTeams?.map((teamSummary) => {
+    return { value: teamSummary.name, label: teamSummary.name };
+  });
+
+  const [defaultTeamName, setDefaultTeamName] = useState<string | undefined>(
+    currentDefaultTeamName
+  );
+
+  const [requestState, setRequestState] = useState<"loading" | undefined>(
+    undefined
+  );
+
+  const onFormSubmit = async (event: FormEvent): Promise<void> => {
+    event.preventDefault();
+    try {
+      setRequestState("loading");
+      const response = await configAPI.update({
+        mdm: { apple_bm_default_team: defaultTeamName },
+      });
+      setRequestState(undefined);
+      onCancel();
```

Comment:
you can move `onCancel()` call outside the try-catch as we always want it to run or put in a `finally` block.